crates/wasmtime/src/runtime/vm/gc/enabled/null.rs

```diff
@@ -202,8 +201,8 @@ unsafe impl GcHeap for NullHeap {
         self.no_gc_count -= 1;
     }
 
-    fn heap_slice(&self) -> &[UnsafeCell<u8>] {
-        let ptr = self.heap.as_ptr().cast();
+    fn heap_slice(&self) -> &[u8] {
+        let ptr = self.heap.as_ptr();
         let len = self.heap.len();
         unsafe { core::slice::from_raw_parts(ptr, len) }
```

Comment:
More drive-by review: perhaps adding an `unsafe fn as_slice(&self)` to `Mmap`? (same for above for the drc heap)